File: frontend/lib/api/system-settings.ts
import { apiClient } from './client';

export interface SystemSetting {
  key: string;
  value: string | number | boolean;
  type: 'string' | 'number' | 'boolean' | 'select';
  label: string;
  description: string;
  category: 'ai' | 'video' | 'notifications' | 'system';
  options?: string[];
  min_value?: number;
  max_value?: number;
  created_at: string;
  updated_at: string;
}

export interface SystemSettingUpdate {
  value: string | number | boolean;
}

export interface SystemSettingsResponse {
  settings: SystemSetting[];
  total: number;
}

export interface SystemStatus {
  websocket_connected: boolean;
  database_connected: boolean;
  ai_model_loaded: boolean;
  file_system_ready: boolean;
  last_updated: string;
}

export interface SystemHealthResponse {
  status: SystemStatus;
  version: string;
  uptime?: string;
}

export type SettingCategory = 'ai' | 'video' | 'notifications' | 'system';

class SystemSettingsApi {
  async getSettings(category?: SettingCategory): Promise<SystemSettingsResponse> {
    const params = new URLSearchParams();
    if (category) {
      params.append('category', category);
    }
    
    const response = await apiClient.get(`/api/v1/system-settings/?${params.toString()}`);
    return response.data;
  }

  async getSetting(key: string): Promise<SystemSetting> {
    const response = await apiClient.get(`/api/v1/system-settings/${key}`);
    return response.data;
  }

  async updateSetting(key: string, updateData: SystemSettingUpdate): Promise<SystemSetting> {
    const response = await apiClient.put(`/api/v1/system-settings/${key}`, updateData);
    return response.data;
  }

  async resetSettings(): Promise<SystemSettingsResponse> {
    const response = await apiClient.post('/api/v1/system-settings/reset');
    return response.data;
  }

  async getSystemHealth(): Promise<SystemHealthResponse> {
    const response = await apiClient.get('/api/v1/system-settings/health/status');
    return response.data;
  }

  // Batch update multiple settings
  async updateMultipleSettings(updates: { key: string; value: string | number | boolean }[]): Promise<SystemSetting[]> {
    const promises = updates.map(update => 
      this.updateSetting(update.key, { value: update.value })
    );
    return Promise.all(promises);
  }
}

export const systemSettingsApi = new SystemSettingsApi();

File: backend/app/models/system_settings.py
from pydantic import BaseModel, Field
from typing import Optional, Union, List
from datetime import datetime
from enum import Enum

class SettingType(str, Enum):
    STRING = "string"
    NUMBER = "number"
    BOOLEAN = "boolean"
    SELECT = "select"

class SettingCategory(str, Enum):
    AI = "ai"
    VIDEO = "video"
    NOTIFICATIONS = "notifications"
    SYSTEM = "system"

class NotificationFrequency(str, Enum):
    IMMEDIATE = "immediate"
    HOURLY = "hourly"
    DAILY = "daily"
    WEEKLY = "weekly"

class LogLevel(str, Enum):
    DEBUG = "DEBUG"
    INFO = "INFO"
    WARNING = "WARNING"
    ERROR = "ERROR"

class SystemSetting(BaseModel):
    key: str = Field(..., description="Unique setting key")
    value: Union[str, int, float, bool] = Field(..., description="Setting value")
    type: SettingType = Field(..., description="Data type of the setting")
    label: str = Field(..., description="Human-readable label")
    description: str = Field(..., description="Setting description")
    category: SettingCategory = Field(..., description="Setting category")
    options: Optional[List[str]] = Field(None, description="Available options for select type")
    min_value: Optional[Union[int, float]] = Field(None, description="Minimum value for number type")
    max_value: Optional[Union[int, float]] = Field(None, description="Maximum value for number type")
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

class SystemSettingUpdate(BaseModel):
    value: Union[str, int, float, bool] = Field(..., description="New setting value")

class SystemSettingsResponse(BaseModel):
    settings: List[SystemSetting]
    total: int

class SystemStatus(BaseModel):
    websocket_connected: bool
    database_connected: bool
    ai_model_loaded: bool
    file_system_ready: bool
    last_updated: datetime = Field(default_factory=datetime.utcnow)

class SystemHealthResponse(BaseModel):
    status: SystemStatus
    version: str = "1.0.0"
    uptime: Optional[str] = None
